### PR TITLE
process IMG eggnog result files

### PR DESCRIPTION
**Pull request by Tianhao-Gu**

### CODE REVIEW

File: scripts/parse_eggnog_result.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+                             aws_access_key_id=ACCESS_KEY,
+                             aws_secret_access_key=SECRET_KEY)
+    try:
+        s3_client.upload_file(str(processed_file), BUCKET, s3_path)
+        print(f"File has been uploaded to s3://{BUCKET}/{s3_path}")
+    except Exception as e:
+        raise ValueError(f"Error uploading {processed_file} to MinIO: {e}")
+
+
+def _validate_directory(data_dir: Path) -> str:
```
Comment (MrCreosote) [inline scripts/parse_eggnog_result.py]: _get_source_file would be a better name

File: scripts/parse_eggnog_result.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    batch_dirs = _get_batch_dirs(RESULT_DIR)
+    total_data_ids, processed_data_ids = 0, 0
+    for batch_dir in batch_dirs:
+        data_ids = [item for item in os.listdir(os.path.join(RESULT_DIR, batch_dir)) if
+                    os.path.isdir(os.path.join(RESULT_DIR, batch_dir, item))]
+        total_data_ids += len(data_ids)
+        for data_id in data_ids:
+            data_dir = RESULT_DIR / batch_dir / data_id
+            if _process_annotation_file(data_dir, data_id):
+                processed_data_ids += 1
```
Comment (MrCreosote) [inline scripts/parse_eggnog_result.py:128]: ideally would exit w/ non zero code if all files aren't processed correctly

File: scripts/parse_eggnog_result.py (hunk 1 above)
Comment (Tianhao-Gu) [reply scripts/parse_eggnog_result.py]: 👍

File: scripts/parse_eggnog_result.py (hunk 2 above)
Comment (Tianhao-Gu) [reply scripts/parse_eggnog_result.py:128]: 👍

**Review by MrCreosote — APPROVED** [review]

**Pull request merged**